partners_api/partners.cpp

```diff
@@ -57,13 +57,16 @@ PartnerInfo::PartnerInfo(int partnerIndex, std::string && name)
 
 std::string const & PartnerInfo::GetBannerPlacementId() const
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunreachable-code"
 #if defined(OMIM_OS_IPHONE)
   return m_iosBannerPlacementId;
 #elif defined(OMIM_OS_ANDROID)
   return m_androidBannerPlacementId;
 #endif
   static std::string kEmptyStr;
   return kEmptyStr;
+#pragma GCC diagnostic pop
 }
```

Comment:
I would prefer to add
#define UNREACHABLE_CODE __builtin_unreachable

And use it here and below like UNREACHABLE_CODE();

Reply:
Let's remove the unreachable code everywhere. It's not needed.